Dockerfile

```diff
@@ -25,15 +22,13 @@ RUN apk add --no-cache libgcc libstdc++ build-base linux-headers
 # Set environment variables
 ENV PYTHONDONTWRITEBYTECODE=1
 ENV PYTHONUNBUFFERED=1
-ENV POETRY_NO_INTERACTION=1 \
-    POETRY_VIRTUALENVS_IN_PROJECT=1 \
-    POETRY_VIRTUALENVS_CREATE=1 \
-    POETRY_CACHE_DIR=/tmp/poetry_cache
+ENV UV_PROJECT_ENVIRONMENT=/app/.venv \
+    UV_CACHE_DIR=/tmp/uv_cache
 
 # Install python dependencies to a virtualenv
-COPY pyproject.toml poetry.lock ./
-RUN pip install poetry=="${poetry_version}"
-RUN poetry install --without dev --no-root && rm -rf $POETRY_CACHE_DIR
+COPY pyproject.toml uv.lock ./
+RUN pip install uv==0.9.5
```

Comment:
Based on https://docs.astral.sh/uv/guides/integration/docker/#installing-uv
we might be able to avoid the pip install here by just copying the binary from an official image